### PR TITLE
[Merged by Bors] - feat(SetTheory/Game/Nim): make `grundyValue` a nimber

### PR DESCRIPTION
**Pull request by vihdzp**
We redefine the grundy value of an impartial game as a nimber. That way, we can prove a more general form of `grundyValue_add`.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Game/Nim.lean [hunk 1]
```diff
@@ -313,68 +317,56 @@ theorem grundyValue_ne_moveRight {G : PGame} [G.Impartial] (i : G.RightMoves) :
...
+@[simp]
+theorem grundyValue_add (G H : PGame) [G.Impartial] [H.Impartial] :
+    grundyValue (G + H) = grundyValue G + grundyValue H := by
+  rw [← (grundyValue G).toOrdinal_toNimber, ← (grundyValue H).toOrdinal_toNimber,
+    ← grundyValue_nim_add_nim, grundyValue_eq_iff_equiv]
+  exact add_congr (equiv_nim_grundyValue G) (equiv_nim_grundyValue H)
+
+theorem grundyValue_add_nat (G H : PGame) [G.Impartial] [H.Impartial] {n m : ℕ}
+    (hG : grundyValue G = ∗n) (hH : grundyValue H = ∗m) : grundyValue (G + H) = ∗(n ^^^ m) := by
+  rw [grundyValue_add, hG, hH, add_nat]
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Game/Nim.lean:369]: This lemma is a bit weirdly stated, no? Do we need it?

Comment (vihdzp) [reply Mathlib/SetTheory/Game/Nim.lean:369]: I don't think we actually need it, I just wanted to highlight that this previous version of the theorem is immediate from the one we have now.

Comment (vihdzp) [reply Mathlib/SetTheory/Game/Nim.lean]: Yeah, I've removed it.

File: Mathlib/SetTheory/Game/Nim.lean [hunk 2]
```diff
@@ -313,68 +318,52 @@ theorem grundyValue_ne_moveRight {G : PGame} [G.Impartial] (i : G.RightMoves) :
...
 
--- Todo: redefine `grundyValue` as a nimber, and prove `grundyValue (nim a + nim b) = a + b` for all
--- nimbers.
-
-/-- The Grundy value of the sum of two nim games with natural numbers of piles equals their bitwise
-xor. -/
-@[simp]
-theorem grundyValue_nim_add_nim (n m : ℕ) : grundyValue (nim.{u} n + nim.{u} m) = n ^^^ m := by
+/-- The Grundy value of the sum of two nim games with natural numbers of piles equals their nimber
+addition. -/
```
Comment (YaelDillies) [inline Mathlib/SetTheory/Game/Nim.lean:340]:
```suggestion
/-- The Grundy value of the sum of two nim games equals their nimber addition. -/
```

**Review by YaelDillies — APPROVED** [review]
maintainer merge

apart from that

**Comment by kim-em** [comment]
bors merge

**Pull request closed without merging**